crates/solvers/src/domain/solution.rs

```diff
@@ -34,33 +34,41 @@ pub enum Trade {
 pub struct Fulfillment {
     order: order::Order,
     executed: U256,
+    fee: Option<U256>,
```

Comment:
I refactored this a bit and added a `solution::Fee` type that models how the fee is charged to the order. It is either a `Protocol` fee, that is, the fee is charged naturally by including the order, or a `Surplus` fee which is an additional surplus fee that the solver can charge.

I used `Surplus` as the name since, from the solver's perspective, the only instance of surplus fees that it chooses are for these kinds of orders. Hope this helps clarify things (and then we have a type that documents what field represents).